serial-vault-services-snap/bin/serviceinit.sh

```diff
@@ -0,0 +1,59 @@
+#!/bin/sh
+
+# Initialize and start the database service
+echo "Initialize the database"
+$SNAP/usr/bin/wrapper-initialize initdb
+$SNAP/usr/bin/wrapper-pg_ctl -D $SNAP_USER_COMMON/data -l $SNAP_USER_COMMON/logs/logfile start
```

Comment:
sorry, I forgot to reply to this. There is indeed a hook to be executed when snap configuration changes. That includes when first install or in any upgrade. You could use it to execute serviceinit. See https://github.com/snapcore/snapd/wiki/hooks
Maybe implement this in another PR, better